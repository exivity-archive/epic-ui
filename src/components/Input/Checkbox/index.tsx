import * as React from 'react'
import styled from 'styled-components'
import { Theme } from '../../../theme';

export interface IProps {
  checked: boolean
  onClick: () => void
  className?: string
  theme: Theme
}

const Checkbox: React.FC<IProps> = ({ checked, onClick, className, theme }) => {
  return <input className={className} type='checkbox' onClick={onClick} defaultChecked={checked} />
}

Checkbox.defaultProps = {
  checked: false
}

/** @component */
export default styled(Checkbox)`
  margin-top: 1px;
  margin-right: ${props => props.theme.global.margin}
  
  /* ${props => props.checked && 'some css'} */
`
