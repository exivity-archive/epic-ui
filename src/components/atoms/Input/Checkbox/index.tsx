import * as React from 'react'

import styled, { css } from 'styled-components'

interface ICheckboxWrapperProps {
  checked: boolean
}

const CheckboxWrapper = styled.div`
    margin: 20px;
    width: 100px;
    height: 100px;
    overflow: hidden;

    ${(props: ICheckboxWrapperProps) => props.checked && css`&:after {
      position: relative;
      left: 5px;
      top: -30px;
      transform: rotateZ(45deg);
      border: solid #fff;
      border-width: 0 3px 3px 0;
      content: ' ';
      display: block;
      width: 6px;
      height: 12px;
    }
    `}
`

export interface ICheckboxProps {
  checked: boolean
  onClick?: () => void
  onChange?: (value: boolean) => void
  className?: string
}

export const Checkbox: React.FC<ICheckboxProps> = ({ checked, onClick, className, onChange }) => {
  return <CheckboxWrapper checked={checked}>
      <input 
        className={className} 
        type='checkbox' 
        onClick={onClick} 
        onChange={onChange ? (e: React.ChangeEvent<HTMLInputElement>) => onChange(e.target.checked) : undefined} 
        defaultChecked={checked} />
      </CheckboxWrapper>
}

export default styled(Checkbox)`
  width: 20px;
  height: 20px;
  cursor: pointer;
  background: ${props => props.checked ? 'rgba(40,40,40,0.7)' : 'rgba(40,40,40,0.2)'};
  color: black;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  position: relative;
  left: -5px;
  top: -5px;

  :focus {
    outline:0;
  }
`
