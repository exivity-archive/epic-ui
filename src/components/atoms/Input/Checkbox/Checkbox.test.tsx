import * as React from 'react'

import { shallow } from 'enzyme'

import Checkbox from '.'

test('renders checkbox', () => {
  const checkbox = shallow(
      <Checkbox checked onClick={() => { return }} />
  )
  expect(checkbox).toMatchSnapshot()
})

test('checked gets flipped in value when checkbox is clicked', () => {
  let checked: boolean = false
  const mockCallBack = jest.fn(() => checked = !checked)

  const checkbox = shallow(<Checkbox checked={checked} onClick={mockCallBack} />)
  
  checkbox.simulate('click')
  expect(mockCallBack.mock.calls.length).toEqual(1)
  // @ts-ignore
  expect(checked === true).toBe(true)
})
