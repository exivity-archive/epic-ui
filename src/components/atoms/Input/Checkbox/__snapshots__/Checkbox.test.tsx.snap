// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checkbox 1`] = `
<input
  className="sc-bdVaJa fkHexh"
  defaultChecked={true}
  onClick={[Function]}
  type="checkbox"
/>
`;
